packages/ai-bot/lib/ai-cost.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  getCurrentActiveSubscription,
+  getUserByMatrixUserId,
+  spendCredits,
+} from '@cardstack/billing/billing-queries';
+import { PgAdapter, TransactionManager } from '@cardstack/postgres';
+import { logger, retry } from '@cardstack/runtime-common';
+import * as Sentry from '@sentry/node';
+
+let log = logger('ai-bot');
+
+export async function saveUsageCost(
+  pgAdapter: PgAdapter,
+  matrixUserId: string,
+  generationId: string,
+) {
+  try {
+    // Generation data is sometimes not immediately available, so we retry a couple of times until we are able to get the cost
+    let costInUsd = await retry(() => fetchGenerationCost(generationId), {
+      retries: 10,
+      delayMs: 500,
+    });
+
+    let creditsConsumed = Math.round(costInUsd / 0.001);
```

Comment:
Make 0.001 a named constant in module scope. CREDITS_PER_USD or something like that?

Reply:
Thanks, done